mapper-runtime/src/main/java/com/datastax/oss/driver/internal/mapper/DaoBase.java

```diff
@@ -113,14 +113,45 @@ protected static SimpleStatement replaceKeyspaceAndTablePlaceholders(
     return SimpleStatement.newInstance(queryString);
   }
 
+  /** Builds an instance of a class referenced in a {@link QueryProvider} annotation. */
+  protected static <ProviderT> ProviderT buildQueryProvider(
+      Class<ProviderT> providerClass,
+      String origin,
+      MapperContext context,
+      EntityHelper<?>... entityHelpers) {
+
+    Class<?>[] parameterTypes = new Class<?>[1 + entityHelpers.length];
+    parameterTypes[0] = MapperContext.class;
+    for (int i = 1; i < parameterTypes.length; i++) {
+      parameterTypes[i] = EntityHelper.class;
+    }
+
+    Object[] arguments = new Object[1 + entityHelpers.length];
+    arguments[0] = context;
+    System.arraycopy(entityHelpers, 0, arguments, 1, entityHelpers.length);
+    try {
+      return providerClass.getDeclaredConstructor(parameterTypes).newInstance(arguments);
```

Comment:
This means that if the constructor does not match what is expected, things will fail at runtime and not at compile time right?  Is there any reason we couldn't create the instance in generated code?

Reply:
🤦‍♂  you're right, we don't need reflection here